build-utils/FindGMT.cmake

```diff
@@ -1,124 +1,109 @@
 #
-#
 # Locate GMT
 #
 # This module accepts the following environment variables:
 #
-#    GMT_DIR or GMT_ROOT - Specify the location of GMT
+# GMT_DIR or GMT_ROOT - Specify the location of GMT
 #
 # This module defines the following CMake variables:
 #
-#    GMT_FOUND - True if libgmt is found
-#    GMT_LIBRARY - A variable pointing to the GMT library
-#    GMT_INCLUDE_DIR - Where to find the headers
-#    GMT_INCLUDE_DIRS - Where to find the headers
-#    GMT_DEFINITIONS - Extra compiler flags
+# GMT_FOUND - True if libgmt is found GMT_LIBRARY - A variable pointing to the
+# GMT library GMT_INCLUDE_DIR - Where to find the headers GMT_INCLUDE_DIRS -
+# Where to find the headers GMT_DEFINITIONS - Extra compiler flags
 
-#=============================================================================
+# =============================================================================
 # Inspired by FindGDAL
 #
 # Distributed under the OSI-approved bsd license (the "License")
 #
-# This software is distributed WITHOUT ANY WARRANTY; without even the
-# implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
-# See COPYING-CMAKE-SCRIPTS for more information.
-#=============================================================================
+# This software is distributed WITHOUT ANY WARRANTY; without even the implied
+# warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See
+# COPYING-CMAKE-SCRIPTS for more information.
+# =============================================================================
 
 # This makes the presumption that you include gmt.h like
 #
-#include "gmt.h"
+# include "gmt.h"
 
-if (UNIX AND NOT GMT_FOUND)
-	# Use gmt-config to obtain the libraries
-	find_program (GMT_CONFIG gmt-config
-		HINTS
-		${GMT_DIR}
-		${GMT_ROOT}
-		$ENV{GMT_DIR}
-		$ENV{GMT_ROOT}
-		PATH_SUFFIXES bin
-		PATHS
-		/sw # Fink
-		/opt/local # DarwinPorts
-		/opt/csw # Blastwave
-		/opt
-		/usr/local
-	)
+if(UNIX AND NOT GMT_FOUND)
+  # Use gmt-config to obtain the libraries
+  find_program(
+    GMT_CONFIG gmt-config
+    HINTS ${GMT_DIR} ${GMT_ROOT} $ENV{GMT_DIR} $ENV{GMT_ROOT}
+    PATH_SUFFIXES bin
+    PATHS /sw # Fink
```

Comment:
On all recent versions of MacOS, Fink has installed to /opt/sw rather than /sw. So perhaps add this path, too (while keeping /sw for those on older versions?)

Reply:
Thank you, seems that @tomoreilly ran into the same problem: https://discourse.cmake.org/t/how-to-direct-findx11-and-findopengl-to-look-in-specific-directory/4790

Using `CMAKE_PREFIX_PATH=/usr/X11` cmake now finds the xquartz version of OpenGL instead of the framework version and thus the headers are in the expected locations now. Next problem is, that cmake does not seem to find any glx libraries so mbview will fail to link because it cannot find `_glXCreateContext` and other glx related symbols. The `OpenGL::GLX` target doesn't even seem to exist on macos... hrm...